src/Database/PostgreSQL/Simple/FromRow.hs

```diff
@@ -450,6 +452,11 @@ instance FromField a => FromRow [a] where
       n <- numFieldsRemaining
       replicateM n field
 
+instance (FromField a) => FromRow (NonEmpty a) where
+    fromRow = do
+      n <- numFieldsRemaining
+      NE.fromList <$> replicateM n field
```

Comment:
I wonder if I shouldn't be using `empty` (`RowParser` has an `Alternative` instance) instead of letting `NE.fromList` throw. Any thought?

Reply:
```haskell
liftA2 (:|) field fromRow
```

Doesn't that work?

Reply:
@phadej Maybe? Since we don't have tests I'm not certain of the behaviour of `fromRow []`. That's what I'm wondering.

Reply:
Maybe we should have tests then!

Reply:
Yep', currently writing some ;)

Reply:
@phadej Hmm, I'm afraid the inner workings of the library are a tad intricate for the uninitiated. 
I'm not sure I'll be able to produce meaningful tests in a reasonable time.